fix(HMS-2428): reduce the bundle size by 50%

amirfefer opened the pull request:
When building this react app, webpack throws a warning regarding bundle size, as you can see in the following screenshot, 2 chunks exceed the limit of 250 KB:

![Screen Shot 2023-08-25 at 12 13 40](https://github.com/RHEnVision/provisioning-frontend/assets/11807069/4f6bf869-4599-4a6a-b3ca-fe92595efcf1)

These two files are related to patternfly-core and patternfly-table. By using treeshaking via babel-transform and code spiting, the bundle is reduced by 50%
![Screen Shot 2023-08-24 at 23 47 59](https://github.com/RHEnVision/provisioning-frontend/assets/11807069/e83c5860-7de8-44c8-bb82-8ee15a7ed0ca)

In addition, this PR adds `analyze` to package.json scripts, in order to build and see a full report.

The main gist of the PR is based on the patterfly-core treeshaking is documented [here](https://github.com/RedHatInsights/frontend-components#treeshaking-pf-with-babel-plugin)

ezr-ondrej commented:
wow, looks to me almost like magic xD would you mind to describe what's been done in the PR description (or even an Markdown file if you want) over what you've done and why it helped? I bet it's gonna be useful in the future and we can use it to advise other teams :)

akhil-jha commented:
/retest

amirfefer commented:
@ezr-ondrej, The `patterfly-core` treeshaking is already documented [here](https://github.com/RedHatInsights/frontend-components#treeshaking-pf-with-babel-plugin)
I wonder why it isn't included in the frontend-starter app already.

On top of it I added code splitting (using `react.lazy`) and also updated related packages.

ezr-ondrej commented:
Ah, too much changes in package-lock in the previous PRs :/
Can you rebase @amirfefer pls? :pray:

ezr-ondrej reviewed (approved):
Thanks, that looks great and works well! :+1:
